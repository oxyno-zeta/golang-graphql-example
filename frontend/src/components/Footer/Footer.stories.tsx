import React from 'react';
import { StoryFn, Meta } from '@storybook/react';
import Footer, { Props } from './Footer';

export default {
  title: 'Components/Footer',
  component: Footer,
} as Meta<typeof Footer>;

export const Playground: StoryFn<typeof Footer> = function C(args: Props) {
  return <Footer {...args} />;
};
