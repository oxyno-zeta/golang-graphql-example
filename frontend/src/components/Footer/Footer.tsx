import React from 'react';
import Typography, { type TypographyProps } from '@mui/material/Typography';
import SvgIcon from '@mui/material/SvgIcon';
import { mdiHeart } from '@mdi/js';
import Box from '@mui/material/Box';
import type { SxProps } from '@mui/material';

export interface Props {
  readonly containerBoxSx?: SxProps;
  readonly typographyProps?: TypographyProps;
}

function Footer({ containerBoxSx = {}, typographyProps = {} }: Props) {
  return (
    <Box
      sx={{
        alignItems: 'center',
        display: 'flex',
        textAlign: 'center',
        flexDirection: 'column',
        margin: '10px 0',
        ...containerBoxSx,
      }}
    >
      <Typography sx={{ display: 'flex' }} {...typographyProps}>
        Todo list application / With{' '}
        <SvgIcon color="error">
          <path d={mdiHeart} />
        </SvgIcon>{' '}
        by Oxyno-zeta
      </Typography>
    </Box>
  );
}

export default Footer;
