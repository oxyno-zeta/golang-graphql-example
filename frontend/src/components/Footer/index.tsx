import Footer, { type Props as FooterProps } from './Footer';

export default Footer;
export type { FooterProps };
