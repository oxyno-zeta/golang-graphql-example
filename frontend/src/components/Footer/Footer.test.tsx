import React from 'react';
import { render } from '@testing-library/react';
// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';

import Footer from './Footer';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

describe('Footer', () => {
  it('should display footer', async () => {
    const { container } = render(<Footer />);
    // Now find errors
    expect(container).toHaveTextContent('Todo list application');
    expect(container).toMatchSnapshot();
  });
});
