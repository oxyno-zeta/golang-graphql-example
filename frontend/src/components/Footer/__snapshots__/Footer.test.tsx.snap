// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should display footer 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-align: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 10px 0;
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  fill: currentColor;
  font-size: 1.5rem;
  color: #d32f2f;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1  emotion-1"
    >
      Todo list application / With
       
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium  emotion-2"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,21.35L10.55,20.03C5.4,15.36 2,12.27 2,8.5C2,5.41 4.42,3 7.5,3C9.24,3 10.91,3.81 12,5.08C13.09,3.81 14.76,3 16.5,3C19.58,3 22,5.41 22,8.5C22,12.27 18.6,15.36 13.45,20.03L12,21.35Z"
        />
      </svg>
       
      by Oxyno-zeta
    </p>
  </div>
</div>
`;
